### PR TITLE
[Multichain] Create global utils directory and store common utils

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/popup/components/InviteItem.vue [hunk 1]
```diff
@@ -87,7 +87,8 @@ import {
 } from '@aeternity/aepp-sdk';
 import { useStore } from 'vuex';
 import { useRouter } from 'vue-router';
-import { APP_LINK_WEB, formatDate } from '../utils';
+import { formatDate } from '@/utils';
+import { APP_LINK_WEB } from '../utils';
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/popup/components/InviteItem.vue:91]: Nope, the new utils are stored in the `src/` directory. The previous one is `src/popup/`. But you're right - it's confusing. I'll update the second path to use `@`.

File: src/protocols/aeternity/utils.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+export function getTransactionTipUrl(transaction: ITransaction): string {
+  return (
+    transaction.tipUrl
+    || transaction.url
+    || (
+      !transaction.pending
+      && !transaction.claim
+      && transaction.tx.log?.[0]
+      && transaction.tx?.function
```
Comment (pstachel) [inline src/protocols/aeternity/utils.ts:108]: I think you should check the presence of `tx` in line 107 - there's no need of doing it in 108

**Review by pstachel — APPROVED** [review]

File: src/protocols/aeternity/utils.ts (hunk 2 above)
Comment (peronczyk) [reply src/protocols/aeternity/utils.ts:108]: `tx` property is required in the `ITransaction` interface. So there's no need to check for it's existance.

File: tests/unit/transaction-details.spec.js [hunk 3]
```diff
@@ -113,7 +113,6 @@ function mountComponent({ hasError = false } = {}) {
       getTx: () => () => testTransaction,
       getTxType: () => () => 'provide liquidity',
       getTxSymbol: () => () => AE_SYMBOL,
-      getTransactionTipUrl: () => () => '',
```
Comment (subhod-i) [inline tests/unit/transaction-details.spec.js:116]: Why?

File: src/utils/crypto.ts
Comment (subhod-i) [inline src/utils/crypto.ts:1]: This is not related to crypto. This is just another helper function that converts a number or string to BigNumber.
The method could be renamed to `toBigNumber` and could be moved to `./common.ts`

File: src/store/plugins/veeValidate.js
Comment (subhod-i) [inline src/store/plugins/veeValidate.js:1]: Can this be moved to `utils/validators.ts` file?

File: src/protocols/aeternity/helpers/index.ts
Comment (subhod-i) [inline src/protocols/aeternity/helpers/index.ts:1]: `helpers` directory has just one file. Instead, Would `aeternity/helpers.ts` file work?

**Review by subhod-i — CHANGES_REQUESTED** [review]

File: tests/unit/transaction-details.spec.js (hunk 3 above)
Comment (peronczyk) [reply tests/unit/transaction-details.spec.js:116]: Because `getTransactionTipUrl` is not anymore present in the store so there is no need to mock it.

File: src/utils/crypto.ts
Comment (peronczyk) [reply src/utils/crypto.ts:1]: Makes sense. Done - I changed the name to `toShiftedBigNumber`.

File: src/store/plugins/veeValidate.js
Comment (peronczyk) [reply src/store/plugins/veeValidate.js:1]: What do you mean by "this"? The whole file?

File: src/protocols/aeternity/helpers/index.ts
Comment (peronczyk) [reply src/protocols/aeternity/helpers/index.ts:1]: For now it has only one file. But another file moves more files here. Like for example the transaction resolvers.

File: src/store/plugins/veeValidate.js
Comment (subhod-i) [reply src/store/plugins/veeValidate.js:1]: Yes.

File: src/protocols/aeternity/helpers/index.ts
Comment (subhod-i) [reply src/protocols/aeternity/helpers/index.ts:1]: Okay.

File: src/store/plugins/veeValidate.js
Comment (peronczyk) [reply src/store/plugins/veeValidate.js:1]: I don't think it's reasonable until we get rid of Vuex. The `veeValidate` file is imported in the `store/index` file and then passed to the `plugins` array of the store. So right now it is a store plugin.
I totally agree we need to move it from the store but not within the scope of the current PR.

Comment (subhod-i) [reply src/store/plugins/veeValidate.js:1]: Sure.

**Review by subhod-i — APPROVED** [review]
